File: groovy-example/src/test/java/com/vocalabs/egtest/example/groovy/DefaultLanguageAsJavaExampleTest.java
package com.vocalabs.egtest.example.groovy;

import org.junit.Test;

import java.lang.reflect.Method;

import static org.junit.Assert.*;

/**
 * Hand-written tests for Example
 */
public class DefaultLanguageAsJavaExampleTest {

    /** This must be updated every time we change ExampleForReadme or if we change how annotations map to tests. */
    @Test
    public void generatedTestsExist() throws Exception {
        Class<?> generatedClass = getClass().getClassLoader().loadClass("com.vocalabs.egtest.example.groovy.DefaultLanguageAsJavaExample$EgTest");
        assertHasTestMethod("testReturns$greet", generatedClass);
        assertHasTestMethod("testReturns$add", generatedClass);
        assertHasTestMethod("testMatch$SIMPLE_EMAIL_RE", generatedClass);
        assertHasTestMethod("testMatch$validEmail", generatedClass);
        assertHasTestMethod("testException$methodWhichCannotHandleNulls", generatedClass);
        assertHasTestMethod("testException$anotherMethodWhichCannotHandleNulls", generatedClass);
        assertHasTestMethod("testReturns$divide", generatedClass);
        assertHasTestMethod("testMatch$startsWithAsciiVowel", generatedClass);
        assertHasTestMethod("testReturns$startsWithAsciiVowel", generatedClass);
        assertHasTestMethod("testException$startsWithAsciiVowel", generatedClass);
    }

    private void assertHasTestMethod(String name, Class<?> generatedClass) throws NoSuchMethodException {
        Method method = generatedClass.getMethod(name);
        assertNotNull(name, method.getAnnotation(Test.class));
    }
}

File: kotlin-example/src/test/java/com/vocalabs/egtest/example/kotlin/DefaultLanguageAsJavaExampleTest.kt
package com.vocalabs.egtest.example.kotlin

import org.junit.Test
import org.junit.Assert.*
import org.junit.Ignore

/**
 * Hand-written tests for Example
 */
class DefaultLanguageAsJavaExampleTest {

    /** This must be updated every time we change ExampleForReadme or if we change how annotations map to tests.  */
    @Test
    @Ignore
    fun generatedTestsExist() {
        fail("Not written")
        /*
        val generatedClass = javaClass.getClassLoader().loadClass("com.vocalabs.egtest.example.kotlin.DefaultLanguageAsJavaExample\$EgTest")
        assertHasTestMethod("testReturns\$greet", generatedClass)
        assertHasTestMethod("testReturns\$add", generatedClass)
        assertHasTestMethod("testMatch\$SIMPLE_EMAIL_RE", generatedClass)
        assertHasTestMethod("testMatch\$validEmail", generatedClass)
        assertHasTestMethod("testException\$methodWhichCannotHandleNulls", generatedClass)
        assertHasTestMethod("testException\$anotherMethodWhichCannotHandleNulls", generatedClass)
        assertHasTestMethod("testReturns\$divide", generatedClass)
        assertHasTestMethod("testMatch\$startsWithAsciiVowel", generatedClass)
        assertHasTestMethod("testReturns\$startsWithAsciiVowel", generatedClass)
        assertHasTestMethod("testException\$startsWithAsciiVowel", generatedClass)
        */
    }

    /*
    private fun assertHasTestMethod(name: String, generatedClass: Class<*>) {
        val method = generatedClass.getMethod(name)
        assertNotNull(name, method.getAnnotation<Test>(Test::class.java!!))
    }
    */
}